lib/apiservers/engine/backends/convert/stats.go

```diff
@@ -316,3 +419,40 @@ func (cs *ContainerStats) currentCPU() {
 	dockerCPU.CPUUsage.TotalUsage += cs.preTotalMhz
 	cs.curDockerStat.CPUStats = dockerCPU
 }
+
+const abc = "abcdefghijklmnopqrstuvxyz"
+
+// toChar will take the provided int and slice the
+// alpha constant to return the string representation.  For example
+// if 1 is provided 'a' will be returned, 2 == 'b', 3 == 'c', etc..
+func toChar(i int) string {
+	return abc[i-1 : i]
+}
+
+// diskMinor will parse the vSphere provided disk name and return the minor id of
+// the disk device.  The func assumes that minor identifiers are multiples of
+// 16 (0,16,32,48,etc)
+func diskMinor(containerID string, name string) uint64 {
+	// split the name to get to the suffix
+	suffix := strings.Split(name, ":")
+	// the int suffix should always be last
+	minor, err := strconv.Atoi(suffix[len(suffix)-1])
+	if err != nil {
+		// log error, but continue and return a minor number of zero
+		// unlikely this would happen, but if it does and there is more than one disk on the vm
+		// then it could go undetected
+		log.Errorf("stats error generating container(%s) disk(%s) minor: %s", containerID, name, err)
+	}
+	// minor identifiers are multiples of 16
+	minor *= 16
+	return uint64(minor)
+}
+
+func createBlkioStatsEntry(minor uint64, op string, value uint64) types.BlkioStatEntry {
```

Comment:
should this have some error handling?  minor and string are generally covered by your own code - i.e. minor is either going to be valid or 0 and string is hard coded to read/write/total in all places, but value does not appear to be guaranteed to be a uint64 necessarily in the case of an error?